Old Apache-commons-net version leads to NPEs

A disposal `Thinghandler` instance for a Smartmeter `Thing` connected via RFC2217 serial implementation leads to:

```
2021-01-16 20:47:01.832 [ERROR] [ding.smartmeter.internal.MeterDevice] - Fatal error occured
java.lang.NullPointerException: null
        at org.apache.commons.net.telnet.TelnetClient._closeOutputStream(TelnetClient.java:84) ~[?:?]
        at org.apache.commons.net.telnet.TelnetOutputStream.close(TelnetOutputStream.java:163) ~[?:?]
        at org.apache.commons.net.telnet.TelnetClient.disconnect(TelnetClient.java:129) ~[?:?]
        at gnu.io.rfc2217.TelnetSerialPort.close(TelnetSerialPort.java:321) ~[?:?]
        at org.openhab.core.io.transport.serial.rxtx.RxTxSerialPort.close(RxTxSerialPort.java:51) ~[?:?]
        at org.openmuc.jrxtx.JRxTxPort.close(JRxTxPort.java:159) ~[bundleFile:?]
        at org.openmuc.j62056.Iec21Port.close(Iec21Port.java:285) ~[bundleFile:?]
        at org.openhab.binding.smartmeter.internal.iec62056.Iec62056_21SerialConnector.closeConnection(Iec62056_21SerialConnector.java:132) ~[bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle$SubscriptionLambdaSubscriber.cancel(FlowableDoOnLifecycle.java:112) [bundleFile:?]
        at io.reactivex.internal.subscribers.BasicFuseableSubscriber.cancel(BasicFuseableSubscriber.java:158) [bundleFile:?]
        at io.reactivex.internal.subscriptions.SubscriptionHelper.cancel(SubscriptionHelper.java:189) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowablePublish$PublishSubscriber.dispose(FlowablePublish.java:183) [bundleFile:?]
        at io.reactivex.internal.disposables.DisposableHelper.dispose(DisposableHelper.java:124) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableRefCount.timeout(FlowableRefCount.java:140) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableRefCount.cancel(FlowableRefCount.java:107) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableRefCount$RefCountSubscriber.cancel(FlowableRefCount.java:229) [bundleFile:?]
        at io.reactivex.internal.subscriptions.SubscriptionArbiter.drainLoop(SubscriptionArbiter.java:218) [bundleFile:?]
        at io.reactivex.internal.subscriptions.SubscriptionArbiter.drain(SubscriptionArbiter.java:187) [bundleFile:?]
        at io.reactivex.internal.subscriptions.SubscriptionArbiter.cancel(SubscriptionArbiter.java:179) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableRepeatWhen$WhenSourceSubscriber.cancel(FlowableRepeatWhen.java:174) [bundleFile:?]
        at io.reactivex.internal.subscriptions.SubscriptionHelper.cancel(SubscriptionHelper.java:189) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.cancel(FlowableSubscribeOn.java:141) [bundleFile:?]
        at io.reactivex.internal.subscriptions.SubscriptionHelper.cancel(SubscriptionHelper.java:189) [bundleFile:?]
        at io.reactivex.internal.subscribers.LambdaSubscriber.cancel(LambdaSubscriber.java:119) [bundleFile:?]
        at io.reactivex.internal.subscribers.LambdaSubscriber.dispose(LambdaSubscriber.java:104) [bundleFile:?]
        at org.openhab.binding.smartmeter.internal.SmartMeterHandler.cancelRead(SmartMeterHandler.java:139) [bundleFile:?]
        at org.openhab.binding.smartmeter.internal.SmartMeterHandler.dispose(SmartMeterHandler.java:128) [bundleFile:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

Thus, this connection becomes a zombie connection that will eventually timeout.

There is a fix for this in apache commons net: https://github.com/apache/commons-net/commit/98d829eb6af5449ac27b66382952de37602ebe4e

and a related fix here:

https://github.com/apache/commons-net/commit/796a288e0c2e17ddf7dc44a4cd790a7eab4e158d

which are both > 3 years old.

The apache-commons-net library was added to openHAB 2 years ago: https://github.com/openhab/openhab-core/pull/694

But the decision was done to use a (at that time already) outdated version.

Is there a reason why we are not using the most recent (3.7.2) one, which is available here? https://mvnrepository.com/artifact/commons-net/commons-net/3.7.2

If there is none, please accept the PR to bump up the version to fix the issue.

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -413,7 +413,7 @@
     <dependency>
       <groupId>commons-net</groupId>
       <artifactId>commons-net</artifactId>
-      <version>3.3</version>
+      <version>3.7.2</version>
       <scope>compile</scope>
     </dependency>
 

--- a/features/karaf/openhab-tp/src/main/feature/feature.xml
+++ b/features/karaf/openhab-tp/src/main/feature/feature.xml
@@ -44,8 +44,8 @@
 	</feature>
 
 	<feature name="openhab.tp-commons-net" description="The Apache Commons Net library" version="${project.version}">
-		<capability>openhab.tp;feature=commons-net;version=3.3</capability>
-		<bundle dependency="true">mvn:commons-net/commons-net/3.3</bundle>
+		<capability>openhab.tp;feature=commons-net;version=3.7.2</capability>
+		<bundle dependency="true">mvn:commons-net/commons-net/3.7.2</bundle>
 	</feature>
 
 	<feature name="openhab.tp-gson" description="Gson" version="${project.version}">

